[tts][ACR-1685] Add description for tts_prepare_sync() (#1592)
Samsung/tizen-docs: docs/application/native/guides/text-input/tts.md

![TTS life-cycle](./media/tts_lifecycle.png)

<a name="parameter_tts"></a>
## TTS parameters

> You can set the following parameters about the TTS:

Reviewer: Ref: It is best not to use "the" in front of tool names.
```suggestion
You can set the following parameters about TTS:
```

  The private data is a setting parameter for applying keys provided by the TTS engine. Using the `tts_set_private_data()` function, you can set the private data as the corresponding key of the TTS engine.

<a name="info_tts"></a>
## TTS information retrieval

> You can get the following information about the TTS:

Reviewer: Ref: It is best not to use "the" in front of tool names.
```suggestion
You can get the following information about TTS:
```

> - [Get the current state](#get) of the TTS. The state is also applied as a precondition for each function.

Reviewer: Ref: It is best not to use "the" in front of tool names.
```suggestion
- [Get the current state](#get) of TTS. The state is also applied as a precondition for each function.
```

> - Get the default voice.

Reviewer: Ref: All lists should be preceded by a “Colon”
```suggestion
- Get the default voice:
```

>   - In the TTS, the voice is defined as a combination of the language and the type, such as male or female.

Reviewer: Ref: It is best not to use "the" in front of tool names.
```suggestion
  - In TTS, the voice is defined as a combination of the language and the type, such as male or female.
```

>   - You can request the synthesis of the text with your own voice option by the parameter of the add text function. However, if you do not set a specific option, the TTS synthesizes the text with the default voice.

Reviewer: Line 65: Should we use code tags for "add text" here to specify the function name?

Author: Yes, I agree your opnion.
It could be better to find out the detail of function. And to maintain consistency dotnet guide, I will use function name itself here.
Thank you.

To enable your application to use the TTS functionality:

1. To use the functions and data types of the TTS (text-to-speech) API (in [mobile](../../api/mobile/latest/group__CAPI__UIX__TTS__MODULE.html) and [wearable](../../api/wearable/latest/group__CAPI__UIX__TTS__MODULE.html) applications), include the `<tts.h>` header file in your application:
   ```cpp
   #include <tts.h>
   ```

2. To use the TTS library, create a TTS handle.

>    The TTS handle is used in other TTS functions as a parameter. After the handle creation, the TTS state changes to `TTS_STATE_CREATED`.

Reviewer: Ref: Code snippets should be preceded by a “colon”.
```suggestion
   The TTS handle is used in other TTS functions as a parameter. After the handle creation, the TTS state changes to `TTS_STATE_CREATED`:
```

>      If you set the engine change callback for the TTS, it is invoked when the engine is changed by the TTS:

Reviewer: Ref: It is best not to use "the" in front of tool names.
```suggestion
     If you set the engine change callback for TTS, it is invoked when the engine is changed by the TTS:
```

>      If you add text in the TTS, that text is handled as an utterance and it obtains its own ID. After you request the TTS process to start, the text is synthesized by an engine and played out. To get a notification when an utterance is started or completed, set the respective callbacks:

Reviewer: Ref: It is best not to use "the" in front of tool names.
```suggestion
     If you add text in TTS, that text is handled as an utterance and it obtains its own ID. After you request the TTS process to start, the text is synthesized by an engine and played out. To get a notification when an utterance is started or completed, set the respective callbacks:
```

- Get the default voice using the `tts_get_default_voice()` function.

>     The TTS synthesizes the text using the default voice, if you do not set the language and the voice type as parameters of the `tts_add_text()` function:

Reviewer: Ref: It is best not to use "the" in front of tool names.
```suggestion
    TTS synthesizes the text using the default voice if you do not set the language and the voice type as parameters of the `tts_add_text()` function:
```

Reviewer: @stom-hwang,
Please remove "The" at the beginning of the sentence. Also, the comma after "voice" is not needed here.
Perhaps you missed it on your last commit.

Reviewer: @stom-hwang,
a gentle reminder to incorporate this change,
Please remove "The" at the beginning of the sentence. Also, the comma after "voice" is not needed here.
Perhaps you missed it on your last commit.
```suggestion
    TTS synthesizes the text using the default voice if you do not set the language and the voice type as parameters of the `tts_add_text()` function:
```

Author: I'm really sorry that I miss it and thank you for reminding me.
I've applied your suggestion right now.
Would you please check again?

```

<a name="prepare"></a>
## Connect and disconnect TTS

> To operate the TTS:

Reviewer: Ref: It is best not to use "the" in front of tool names.
```suggestion
To operate TTS:
```

   }
   ```

   > [!NOTE]
   > If you get the error callback after calling the `tts_prepare()` function, TTS is not available.

2. If you want to connect the daemon synchronously, use the `tts_prepare_sync()` function.

>    If the function returns no error, connection succeeds and the TTS state changes to `TTS_STATE_READY` immediately:

Reviewer: Ref: An article "the" was missing from the sentence.
```suggestion
   If the function returns no error, the connection succeeds and the TTS state changes to `TTS_STATE_READY` immediately:
```

   }
   ```

## Set and get options about the TTS engine

To set and get the options about the TTS engine:

- Set the credential.

>   The credential is a key to verify the authorization about using the TTS engine. The necessity of the credential depends on the engine. If the engine requests the credential, you can set it using the `tts_set_credential()` function:

Reviewer: Addressed a slight grammatical issue in the sentence.
```suggestion
  The credential is a key used to verify the authorization when using the TTS engine. The necessity of the credential depends on the engine. If the engine requests the credential, you can set it using the `tts_set_credential()` function:
```

>   The private data is a setting parameter for applying keys provided by the TTS engine. Using the `tts_set_private_data()` function, you can set the private data and use the corresponding key of the engine. To get the private data which corresponds to a specific key from the engine, use the `tts_get_private_data()` function.

Reviewer: Ref: Code snippets should be preceded by a “colon”.
```suggestion
  The private data is a setting parameter for applying keys provided by the TTS engine. Using the `tts_set_private_data()` function, you can set the private data and use the corresponding key of the engine. To get the private data that corresponds to a specific key from the engine, use the `tts_get_private_data()` function:
```

  ```

<a name="text"></a>
## Add text

> There are defined values, `TTS_VOICE_TYPE_AUTO` and `TTS_VOICE_SPEED_AUTO` for following the default TTS setting. Those values, and the minimum and maximum limits for the speed, are defined in the `tts.h` header file.

Reviewer: A slight grammatical error was fixed.
```suggestion
There are defined values, `TTS_VOICE_TYPE_AUTO` and `TTS_VOICE_SPEED_AUTO`, for following the default TTS setting. Those values, and the minimum and maximum limits for the speed, are defined in the `tts.h` header file.
```

Reviewer: @stom-hwang,
We added a comma at the end of `TTS_VOICE_SPEED_AUTO`, perhaps you missed it on your last commit.

  ```

<a name="control"></a>
## Control playback

To start, pause, and stop the playback:

- To start synthesizing the text added in the queue and play the resulting sound data in sequence, use the `tts_play()` function.

  The state is changed to `TTS_STATE_PLAYING`, and the playback continues until you call the `tts_stop()` or the `tts_pause()` function.

>   If there is no text in the queue, TTS stays in the `TTS_STATE_PLAYING` state for text to be added. In that case, when text is added, TTS starts synthesizing and playing it immediately. There is no need to change the state to `TTS_STATE_READY` by using the `tts_stop()` function even when there is no text in the queue. It will continue to stay in the `TTS_STATE_PLAYING` state.

Reviewer: Ref: Code snippets should be preceded by a “colon”.
```suggestion
  If there is no text in the queue, TTS stays in the `TTS_STATE_PLAYING` state for text to be added. In that case, when text is added, TTS starts synthesizing and playing it immediately. There is no need to change the state to `TTS_STATE_READY` by using the `tts_stop()` function even when there is no text in the queue. It will continue to stay in the `TTS_STATE_PLAYING` state:
```

  ```

> ## Related Information

Reviewer: Ref; For second heading, please use sentence style capitalization.
```suggestion
## Related information
```
